src/cppsim/gate_noisy_evolution.hpp

```diff
@@ -202,12 +202,22 @@ class ClsNoisyEvolution : public QuantumGateBase {
     ClsNoisyEvolution(Observable* hamiltonian,
         std::vector<GeneralQuantumOperator*> c_ops, double time,
         double dt = 1e-6) {
-        _hamiltonian = dynamic_cast<Observable*>(hamiltonian->copy());
+        _hamiltonian = hamiltonian->copy();
         for (auto const& op : c_ops) {
             _c_ops.push_back(op->copy());
             _c_ops_dagger.push_back(op->get_dagger());
         }
-        _effective_hamiltonian = hamiltonian->copy();
+
+        // HermitianQuantumOperatorのチェックを回避するため、
+        // GeneralQuantumOperatorに作り直す。（GeneralQuantumOperatorへのキャストではダメ）
+        // _effective_hamiltonian =
+        //     dynamic_cast<GeneralQuantumOperator*>(hamiltonian->copy());
```

Comment:
このコメントでは、
- 何のチェックが走っているのか
- なぜGeneralQuantumOperatorへのキャストではダメなのか
がわかりません。
その部分を書くとより良いコメントになると思います。

```suggestion
        // HermitianQuantumOperatorは、add_operatorの呼び出し時に、
        // 追加するPauliOperatorがHermitianであるかのチェックが入る。
        // _effective_hamiltonianに追加するPauliOperatorはHermitianとは限らないので、
        // チェックに失敗してしまう可能性がある。
        // したがって、このチェックを回避するためにGeneralQuantumOperatorを生成し、
        // hamiltonianの中身をコピーする。
        //
        // HermitianQuantumOperatorをGeneralQuantumOperatorにキャストする方法では、
        // インスタンスがHermitianQuantumOperatorのままであるため、
        // HermitianQuantumOperator側のadd_operatorが呼び出されてしまい、問題が解決できない。
        // したがって、GeneralQuantumOperatorを実際に作成する必要がある。
```